qualtran/bloqs/arithmetic/bitwise.py

```diff
@@ -221,6 +221,11 @@ def wire_symbol(
 
         return TextBox("~x")
 
+    def on_classical_vals(self, x: 'ClassicalValT') -> Dict[str, 'ClassicalValT']:
+        if hasattr(self.dtype, 'bitsize'):
```

Comment:
it's not obvious to me what this conditional is for. Every data type has `num_qubits`, which could conceivably be used here. If this simplification only works for specific data types, i'd prefer affiratively listing them with `isinstance` rather than using `hasattr`

Reply:
can you add a test for classical action on this bloq

Reply:
added for this and for the other bloq (MultiTargetCNOT)